modules/frontend/search_streaming.go

```diff
@@ -287,22 +356,33 @@ func (s *streamingSearcher) handle(r *http.Request, forwardResults func(*tempopb
 			return ctx.Err()
 		// stream results as they come in
 		case <-time.After(500 * time.Millisecond):
-			p := progress.Load()
-			if p == nil {
-				continue
-			}
+			snap := mProgress.Snapshot() // safe copy of progress list
+			for _, progress := range snap {
+				p := progress.Load()
+				if p == nil {
+					continue
+				}
 
-			result := (*p).result()
-			if result.err != nil || result.statusCode != http.StatusOK { // ignore errors here, we'll get them in the resultChan
-				continue
-			}
+				result := (*p).result()
+				if result.err != nil || result.statusCode != http.StatusOK { // ignore errors here, we'll get them in the resultChan
+					continue
+				}
 
-			err := forwardResults(result.response)
-			if err != nil {
-				level.Error(s.logger).Log("msg", "search streaming: send failed", "err", err)
-				return fmt.Errorf("search streaming send failed: %w", err)
+				// send response to client
+				// FIXME: this will ping pong between each tenant's metrics and client will not get accurate metrics
+				// -- metrics when results are streaming back in??
+				// {"metrics":{"inspectedBytes":452084,"totalBlocks":1,"completedJobs":2,"totalJobs":2,"totalBlockBytes":535418}}
+				// {"metrics":{"inspectedBytes":452945,"totalBlocks":1,"completedJobs":2,"totalJobs":2,"totalBlockBytes":506656}}
+				// {"metrics":{"inspectedBytes":452084,"totalBlocks":1,"completedJobs":2,"totalJobs":2,"totalBlockBytes":535418}}
+				// {"metrics":{"inspectedBytes":452945,"totalBlocks":1,"completedJobs":2,"totalJobs":2,"totalBlockBytes":506656}}
+				// FIXME: need to merge metrics from all tenants and modify response with merged metrics.
+				err := forwardResults(result.response)
```

Comment:
this seems like it will forward each tenant's response individually. we should combine results before returning.

i think you can just call `mProgress.finalResults()` and return that

Reply:
finalResults will not return finalResults, because it calls finalResult on diff.

we want to call result in each diff and return that, I am adding a method on multiProcess called result, will call that here. 

will merge metrics as well, and that should fix the ping ping issue with metrics.

Reply:
this is cleaned up and fixed.